File: modules/client/src/main/java/com/eventhorizon/pulsar/client/factory/RedisConnectionFactory.java
package com.eventhorizon.pulsar.client.factory;

import redis.clients.jedis.DefaultJedisClientConfig;
import redis.clients.jedis.HostAndPort;
import redis.clients.jedis.JedisPool;

public final class RedisConnectionFactory {

    private static final DefaultJedisClientConfig DEFAULT_JEDIS_CONFIG =
            DefaultJedisClientConfig.builder()
                    .clientName("pulsar-client")
                    .build();

    public static JedisPool create(String redisUri) {
        return new JedisPool(
                HostAndPort.from(redisUri),
                DEFAULT_JEDIS_CONFIG
        );
    }
}

File: modules/client/src/main/kotlin/com/eventhorizon/pulsar/client/factory/RedisConnectionFactory.kt
package com.eventhorizon.pulsar.client.factory

import redis.clients.jedis.DefaultJedisClientConfig
import redis.clients.jedis.HostAndPort
import redis.clients.jedis.JedisPool

internal object RedisConnectionFactory {
    private val DEFAULT_JEDIS_CONFIG = DefaultJedisClientConfig.builder()
        .clientName("pulsar-client")
        .build()

    fun create(redisUri: String) : JedisPool {
        return JedisPool(
            HostAndPort.from(redisUri),
            DEFAULT_JEDIS_CONFIG
        )
    }
}